rosidl_typesupport_connext_cpp/resource/srv__type_support.cpp.template

```diff
@@ -95,21 +156,73 @@ int64_t send_request__@(spec.srv_name)(
 void * create_replier__@(spec.srv_name)(
   void * untyped_participant,
   const char * service_name,
-  const void * untyped_datareader_qos,
-  const void * untyped_datawriter_qos,
+  const void * untyped_replier_qos,
   void ** untyped_reader)
 {
-  if (!untyped_participant || !service_name || !untyped_reader) {
+  if (!untyped_participant || !service_name || !untyped_reader || !untyped_replier_qos) {
     return NULL;
   }
 
   DDSDomainParticipant * participant = static_cast<DDSDomainParticipant *>(untyped_participant);
-  const DDS_DataReaderQos * datareader_qos = static_cast<const DDS_DataReaderQos *>(untyped_datareader_qos);
-  const DDS_DataWriterQos * datawriter_qos = static_cast<const DDS_DataWriterQos *>(untyped_datawriter_qos);
+  const rmw_qos_profile_t * replier_qos = static_cast<const rmw_qos_profile_t *>(untyped_replier_qos);
+  DDS_DataReaderQos datareader_qos;
+  DDS_DataWriterQos datawriter_qos;
+
+  if (!get_datareader_qos(participant, datareader_qos)) {
+    RMW_SET_ERROR_MSG("Unable to initialize datareader QoS");
```

Comment:
It looks like these error messages are being overwritten afterwards if the function return NULL.